pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/AggregateOperator.java

```diff
@@ -308,6 +303,20 @@ public Object merge(Object agg, Object value) {
     }
   }
 
+  private static class MergeCounts implements Merger {
+
+    @Override
+    public Object initialize(Object other, DataSchema.ColumnDataType dataType) {
+      return other == null ? 0 : 1;
+    }
+
+    @Override
+    public Object merge(Object left, Object ignored) {
+      // TODO: COUNT(*) doesn't need to parse right object until we support NULL
+      return ((Number) left).doubleValue() + 1;
```

Comment:
this should be supported right?
```suggestion
      return ((Number) left).doubleValue() + (ignored == null ? 0 : 1);
```
and change `ignored` to `right`

Reply:
done